spotify_client.py

```diff
@@ -0,0 +1,70 @@
+from dataclasses import dataclass
+from typing import List, Optional
+import spotipy
+from spotipy.oauth2 import SpotifyOAuth
+from rich.console import Console
+from config import SpotifyConfig
+
+console = Console(force_terminal=True)
+
+
+@dataclass
+class SpotifyTrack:
+    id: str
+    name: str
+    artist: str
+
+
+class SpotifyClient:
+    def __init__(self, config: SpotifyConfig):
+        self.sp = spotipy.Spotify(
+            auth_manager=SpotifyOAuth(
+                client_id=config.client_id,
+                client_secret=config.client_secret,
+                redirect_uri=config.redirect_uri,
+                scope="playlist-modify-public playlist-modify-private",
+            )
+        )
+        self.user_id = self.sp.current_user()["id"]
+
+    def create_playlist(
+        self, name: str, description: str = "", public: bool = True
+    ) -> dict | None:
```

Comment:
When I run client I've got:
```bash
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

this is due to the fact that pycharm installed virtual env with python 3.9 (that is ok at least from what we can read in the README.md). In old pythons you can't have such a typing, you have to have Optional[Dict]. I suggest to use poetry and specify python versions and then use pylint and mypy to ensure typing and other things are correct. poetry will resolve right dependencies versions for the right python (btw you can specify range of python sin the spotify) so it would work correctly in any scenario.

Reply:
There are more typing errors like this but I will not repeat his comment.